server/src/test/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapperTests.java

```diff
@@ -8,93 +8,217 @@
+            }
+            b.endObject().endObject();
+        }));
+
+        DocumentParsingException e = expectThrows(
+            DocumentParsingException.class,
+            () -> mapper.parse(source(b -> b.startObject("field").field("foo", Arrays.asList(10, 20)).endObject()))
+        );
+        assertEquals(
+            "[sparse_vector] fields take hashes that map a feature to a strictly positive float, but got unexpected token " + "START_ARRAY",
+            e.getCause().getMessage()
+        );
+
+        e = expectThrows(DocumentParsingException.class, () -> mapper.parse(source(b -> {
+            b.startArray("foo");
+            {
+                b.startObject().startObject("field").field("bar", 10).endObject().endObject();
+                b.startObject().startObject("field").field("bar", 20).endObject().endObject();
+            }
+            b.endArray();
+        })));
+        assertEquals(
+            "[sparse_vector] fields do not support indexing multiple values for the same rank feature [foo.field.bar] in "
+                + "the same document",
+            e.getCause().getMessage()
         );
-        assertThat(e.getMessage(), containsString(SparseVectorFieldMapper.ERROR_MESSAGE));
     }
 
-    public void testSparseVectorWith7xIndex() throws Exception {
-        IndexVersion version = IndexVersionUtils.randomPreviousCompatibleVersion(random(), IndexVersion.V_8_0_0);
-        Settings settings = Settings.builder().put(IndexMetadata.SETTING_VERSION_CREATED, version).build();
+    public void testCannotBeUsedInMultiFields() {
+        Exception e = expectThrows(MapperParsingException.class, () -> createMapperService(fieldMapping(b -> {
+            b.field("type", "keyword");
+            b.startObject("fields");
+            b.startObject("feature");
+            b.field("type", "sparse_vector");
+            b.endObject();
+            b.endObject();
+        })));
+        assertThat(e.getMessage(), containsString("Field [feature] of type [sparse_vector] can't be used in multifields"));
+    }
 
-        IndexService indexService = createIndex("index", settings);
-        MapperService mapperService = indexService.mapperService();
+    @Override
+    protected Object generateRandomInputValue(MappedFieldType ft) {
+        assumeFalse("Test implemented in a follow up", true);
+        return null;
+    }
 
-        BytesReference mapping = BytesReference.bytes(
-            XContentFactory.jsonBuilder()
-                .startObject()
-                .startObject("_doc")
-                .startObject("properties")
-                .startObject("my-vector")
-                .field("type", "sparse_vector")
-                .endObject()
-                .endObject()
-                .endObject()
-                .endObject()
-        );
+    @Override
+    protected boolean allowsNullValues() {
+        return false;       // TODO should this allow null values?
```

Comment:
Nitpick: Remove TODO before merging

Reply:
Copied / pasted from `RankFeaturesFieldMapper`, so I think it makes sense for re-evaluating in the future.